### PR TITLE
Make all Loggers optional

### PR DESCRIPTION
**Pull request by gwynne**
- All uses of `Logger` throughout the package are now optional for better control over logging.
- Some minor improvements to the unit tests.

_Note: Not an API break because making the inputs optional does not cause any previously valid input to be rejected._

### CODE REVIEW

**Comment by gwynne** [comment]
Tests are failing because the `vapor/swift:5.2` image is badly out of date at this point.

File: Sources/PostgresNIO/Connection/PostgresConnection+Authenticate.swift [hunk 1]
```diff
@@ -7,7 +7,7 @@ extension PostgresConnection {
         username: String,
         database: String? = nil,
         password: String? = nil,
-        logger: Logger = .init(label: "codes.vapor.postgres")
+        logger: Logger? = .init(label: "codes.vapor.postgres")
```
Comment (tanner0101) [inline Sources/PostgresNIO/Connection/PostgresConnection+Authenticate.swift]: Should this be `= nil`? 🤔

File: Sources/PostgresNIO/Connection/PostgresConnection+Database.swift [hunk 2]
```diff
@@ -3,7 +3,7 @@ import Logging
 extension PostgresConnection: PostgresDatabase {
     public func send(
         _ request: PostgresRequest,
-        logger: Logger
+        logger: Logger?
```
Comment (tanner0101) [inline Sources/PostgresNIO/Connection/PostgresConnection+Database.swift]: Same question here re: `= nil`.

File: Sources/PostgresNIO/Connection/PostgresConnection+Authenticate.swift (hunk 1 above)
Comment (gwynne) [reply Sources/PostgresNIO/Connection/PostgresConnection+Authenticate.swift]: Well, it's still nice to have it by default. My intent was never to remove the ability to see what's happening by default, and there are plenty of database messages you _definitely_ want to see in the console if they happen. My expectation for the optionality of it is to have an easy way to silence the logging in cases like unit tests where the `.trace` logging level is likely to be used by the top-level application and scrolling through reports of every last `PostgresMessage` on the connection is tedious 🙂. If either the default `LogHandler` or `console-kit`'s permitted filtering by category, I would use that, but adding that capability is _highly_ nontrivial and doesn't negate the separate uses for this change.

File: Sources/PostgresNIO/Connection/PostgresConnection+Database.swift (hunk 2 above)
Comment (gwynne) [reply Sources/PostgresNIO/Connection/PostgresConnection+Database.swift]: Yeah, for this method it probably should default to `nil` if unspecified.

**Comment by gwynne** [comment]
I withdraw this PR; the complete set of changes needed to make it actually work and viable reaches through `async-kit`, `sql-kit`, `fluent-kit`, etc., and it can't be reasonably done in a source-compatible fashion. If we were a month from GM, maybe, but not now.

**Comment by gwynne** [comment]
I will open a separate PR with the changes for the tests split out.

**Pull request closed without merging**